clippy_lints/src/manual_slice_size_calculation.rs

```diff
@@ -42,15 +45,21 @@ impl<'tcx> LateLintPass<'tcx> for ManualSliceSizeCalculation {
         if !in_constant(cx, expr.hir_id)
             && let ExprKind::Binary(ref op, left, right) = expr.kind
             && BinOpKind::Mul == op.node
-            && let Some(_receiver) = simplify(cx, left, right)
+            && let Some(receiver) = simplify(cx, left, right)
         {
-            span_lint_and_help(
+            let ctxt = expr.span.ctxt();
+            let mut app = Applicability::MachineApplicable;
+            let val_name = snippet_with_context(cx, receiver.span, ctxt, "slice", &mut app).0;
+
+            span_lint_and_sugg(
                 cx,
                 MANUAL_SLICE_SIZE_CALCULATION,
-                expr.span,
-                "manual slice size calculation",
-                None,
-                "consider using std::mem::size_of_val instead");
+                    expr.span,
+                    "manual slice size calculation",
+                    "try",
+                    format!("std::mem::size_of_val({val_name})"),
```

Comment:
we should use `clippy_utils::std_or_core` here (see the ALMOST_SWAPPED lint for an example)